### DISCUSSION TITLE
isReady documentation is unclear and merging query and route params can conflict

### DISCUSSION THREAD

**Discussion by eric-burel** [vercel/next.js#32229] [Ideas]
### Describe the feature you'd like to request

At the time of writing documentation says: 
"
isReady: boolean - Whether the router fields are updated client-side and ready for use. Should only be used inside of useEffect methods and not for conditionally rendering on the server.
"
This is not really true: when in fallback mode, `router.query` will not be fed with route parameters (not query params like foo?bar, but actual route params like /foo/[bar]) until `router.isFallback` becomes false.

### Describe the solution you'd like

I didn't find the right phrasing, but it should be clearer from reading this API doc that if you want to access the route params from `router.query`, you may also want to wait for `router.isFallback` to become false.

**Also** merging route params and query params into a `router.query` object could lead to conflicts, imagine a param being named `documentId` or a query parameters whose name is left free to the user => it could collide with a route parameters. All the more that query params are not accepted as `getStaticPaths` parameters while route params are, so they have a pretty different behaviour in Next.

### Describe alternatives you've considered

Maybe renaming/introducing new better named fields: `isQueryParamsReady` instead of `isReady` for instance that is quite vague.

**Comment by jashwant** [comment]
I came to raise this issue and found yours.

I have created a [codesandbox](https://codesandbox.io/s/snowy-tdd-vd7hy?file=/pages/posts/%5Bid%5D.js) to reproduce the issue.
